components/forwarder/envelope.go

```diff
@@ -0,0 +1,74 @@
+package forwarder
+
+import (
+	"encoding/json"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/pkg/errors"
+)
+
+// messageEnvelope wraps Watermill message and contains destination topic.
+type messageEnvelope struct {
+	DestinationTopic string `json:"destination_topic"`
+
+	UUID     string            `json:"uuid"`
+	Payload  []byte            `json:"payload"`
+	Metadata map[string]string `json:"metadata"`
+}
+
+func newMessageEnvelope(destTopic string, uuid string, payload []byte, metadata map[string]string) (*messageEnvelope, error) {
```

Comment:
maybe you can pass watermill message instead of X params? ;)

Reply:
True: 40e39a7